### DISCUSSION TITLE
Darstellungsfilter mit Auswahlliste aus Dienst

### DISCUSSION THREAD

**Discussion by marko-jer** [e-netze/webgis-community#145] [General]
Ich habe einen Darstellungsfilter angelegt, der die Auswahlliste aus dem Dienst beziehen soll.
Schritt 1 - Layerauswahl:
![image](https://github.com/gis-eni/webgis-forum/assets/1842500/b6e86c19-2bea-403c-aafe-a7ae3348dda7)

Schritt 2 - Auswahlliste lt. Anleitung:
![image](https://github.com/gis-eni/webgis-forum/assets/1842500/96a58b49-203d-40c8-9c0c-7312aa74b045)
![image](https://github.com/gis-eni/webgis-forum/assets/1842500/8d95915d-efc7-45d2-8a36-b059ae2daa5f)
 
und wenn ich den Darstellungsfilter im webGIS  dann aktivieren will, bekomme ich folgende Fehlermeldung:
![image](https://github.com/gis-eni/webgis-forum/assets/1842500/b82ceb88-c8c9-4928-b445-4fb6a74f346d)

Bei "Suchen" funktioniert diese Vorgangsweise für die Auswahlliste ausgezeichnet!

**Comment by jugstalt** [comment]
Ich seh in der Parametrierung keinen Fehler. Fehler 400 ist auch immer blöd, weil das Allegemeiner (SQL) Fehler ist, den der AGS zurück gibt.
Da müsste ich tiefer in den Dienst eintauchen ;) 
Vielleicht kannst du mir den Link zum Dienst schicken (E-Mail) und noch einmal beschreiben, um welches Thema im Dienst es geht, damit ich das nachbauen kann.

**Comment by jugstalt** [comment]
Das Problem ist hier, dass WebGIS das SQL Statement im Filter nicht auf die Logik des Statements parsen kann. Das kann sehr kompliziert werden...
Es werden alle Platzhalter gesucht und dann die Maske gebaut: Für jeden Platzhalter ein Eingabefeld.
Wenn du jetzt die Auswahlliste über den Dienst holst (Distinct Query), muss natürlich bekannt sein, auf welches Feld es sich bezieht. Da, wie gesagt das Statement nicht als solches geparsed wird und WebGIS nicht erkennt, dass [Prompt] sich eigentlich auf das NAME Feld im Dienst bezieht, kommt hier ein Fehler. Wenn du das so machen willst, müsstest du den [Prompt] gleich benennen wie das Feld, auf das der Distinct angewendet wird, also zB:

Filter: 

``NAME='[NAME]'``

**Reply by marko-jer** [reply]
Verstehe, kein Vorteil ohne Nachteil - denn dann muss man den Spaltennamen der DB-Tabelle einsetzten, der ab und zu nicht sprechend ist.

**Comment by DORIS-LandOoe** [comment]
Hallo Jürgen!

Im Viewer wird beim mir bereits die Werteauswahlliste angezeigt. Jedoch erhalte ich bei Auswahl der Werte unterschiedliche Antworten/Fehler.

Hier meine Konfiguration:
Darstellungsfilter:
![image](https://github.com/user-attachments/assets/e74ce0a7-1b51-458c-9833-68edffb00de6)

Auswahlliste:
![image](https://github.com/user-attachments/assets/a1adae21-417a-4e94-a23f-b6b704f8d72f)

Aussehen im Viewer:
![image](https://github.com/user-attachments/assets/eface510-22e0-4cd0-9331-ebb8c43565fd)


Bei Auswahl von "Freilichtmuseum" oder "Spezialmuseum" erhalte ich eine leere Karte - also alle Features werden weggefiltert.
Wenn ich einen Eintrag bestehend aus mehreren Wörtern auswähle, dann bekomme ich die Fehlermeldung: 
04.06.2025 07:45:03 Warning: Error:400
Invalid 'layerDefs' is specified

Muss ich noch etwas berücksichtigen oder funktioniert die Auswahlliste nur per DB-Abfrage?
LG
Markus

**Reply by gis-eni** [reply]
Wenn es sich um ein String Feld in der Datenbank handelt, musst du das auch im SQL Statement berücksichtigen. WebGIS nimmt das Statement nur und ersetzt den Platzhalter durch den Wert: 

``Museumskategorie = Biographisches Museum``

Das wird nicht so nicht hinhauten... richtig ist meiner Meinung nach:

``Museumskategorie = 'Biographisches Museum'``

Also im CMS muss die WHERE-Klausel auch richtig parametriert werden:

``Museumskategorie = '[Museumskategorie]'``
